src/components/ContentNode/CodeListing.vue

```diff
@@ -122,6 +157,21 @@ export default {
         line === '' ? '\n' : line
       ));
     },
+    copyCodeToClipboard() {
+      navigator.clipboard.writeText(this.copyableText)
+        .then(() => {
+          this.copyState = CopyState.success;
+        })
+        .catch((err) => {
+          console.error('Failed to copy text: ', err);
+          this.copyState = CopyState.failure;
+        })
+        .finally(() => {
+          setTimeout(() => {
+            this.copyState = CopyState.idle;
+          }, 1000);
+        });
+    },
```

Comment:
```suggestion
    },
    CopyState,
```

Then, you can also use this constant in the template instead of the literal strings like this:

```
<CopyIcon v-if="copyState === CopyState.idle" ...
```
(etc)

Reply:
That was my bad—thought I was pointing to a line at the same level of `methods` and not inside it. The way you did it works fine, thanks.